aten/src/ATen/native/cpu/SpmmReduceKernel.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <ATen/core/Tensor.h>
+#include <ATen/native/DispatchStub.h>
+
+namespace at { namespace native {
+
+enum SPMM_REDUCE_OP {SPMM_SUM, SPMM_MAX, SPMM_MIN, SPMM_MEAN};
+
+using spmm_reduce_fn = void(*)(const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, SPMM_REDUCE_OP op);
+using spmm_reduce_arg_fn = void(*)(const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, SPMM_REDUCE_OP op);
+using spmm_reduce_backward_input_fn = void(*)(const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, SPMM_REDUCE_OP op);
+using spmm_reduce_backward_input_arg_fn = void(*)(const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, SPMM_REDUCE_OP op);
+using spmm_reduce_backward_weight_fn = void(*)(const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, const Tensor&, SPMM_REDUCE_OP op);
+
+DECLARE_DISPATCH(spmm_reduce_fn, spmm_reduce_stub);
+DECLARE_DISPATCH(spmm_reduce_arg_fn, spmm_reduce_arg_stub);
+DECLARE_DISPATCH(spmm_reduce_backward_input_fn, spmm_reduce_backward_input_stub);
+DECLARE_DISPATCH(spmm_reduce_backward_input_arg_fn, spmm_reduce_backward_input_arg_stub);
+DECLARE_DISPATCH(spmm_reduce_backward_weight_fn, spmm_reduce_backward_weight_stub);
+DECLARE_DISPATCH(spmm_reduce_backward_input_arg_fn, spmm_reduce_backward_weight_arg_stub);
+
+#define AT_DISPATCH_REDUCTION_TYPES(op, ...)                                   \
```

Comment:
Is that something that can be shared across different ops, e.g., `scatter_reduce` or `segment_reduce`? In this case, we should move this to a different file and rename to `REDUCE_OP`?

Reply:
will do

Reply:
Unify with `ReductionType` for `scatter_reduce`, `spmm_reduce` and `segment_reduce`